linode/instancereservedipassignment/framework_schema.go

```diff
@@ -0,0 +1,113 @@
+package instancereservedipassignment
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/listplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/linode/terraform-provider-linode/v2/linode/instancenetworking"
+)
+
+var frameworkResourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Description: "The ID of the IPv4 address, which will be IPv4 address itself.",
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"linode_id": schema.Int64Attribute{
+			Description: "The ID of the Linode to allocate an IPv4 address for.",
+			Required:    true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.RequiresReplace(),
+			},
+		},
+		"public": schema.BoolAttribute{
+			Description: "Whether the IPv4 address is public or private.",
+			Default:     booldefault.StaticBool(true),
+			Computed:    true,
+			Optional:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.RequiresReplace(),
+			},
+		},
+
+		"address": schema.StringAttribute{
+			Description: "The resulting IPv4 address.",
+			Required:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"gateway": schema.StringAttribute{
+			Description: "The default gateway for this address",
+			Computed:    true,
+		},
+		"prefix": schema.Int64Attribute{
+			Description: "The number of bits set in the subnet mask.",
+			Computed:    true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.UseStateForUnknown(),
+			},
+		},
+		"rdns": schema.StringAttribute{
+			Description: "The reverse DNS assigned to this address.",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
```

Comment:
It looks like RDNS can be changed in Update. Should we not preserve its value from state if it leaves empty? User may intend to remove it.